### PR TITLE
Migrate from yarn to pnpm

### PR DESCRIPTION
**Pull request by vscav**
# Description

`pnpm` is used instead of `yarn` to manage the project.

# Changes

- `pnpm` is used instead of `yarn`
  - "workspaces" field is removed from package.json because it is not supported by pnpm - we use a pnpm-workspace.yaml file instead
  - `yarn` is removed from Volta and `pnpm` is not added because [support is currently experimental](https://docs.volta.sh/advanced/pnpm)
  - ` NullVoxPopuli/action-setup-pnpm` is used in our CI to install `pnpm`
  - Node version on CI is upgraded to `16.x`  due to pnpm version requirements
- Dev dependencies of `test-app` are upgraded
- `@ember/string` is added and `ember-export-application-global` is removed (it seems there is an Ember version incompatibility) from `test-app` dev dependencies
- `ember-template-lint` is upgraded because `--no-error-on-unmatched-pattern` CLI option was released in [4.4.1](https://github.com/ember-template-lint/ember-template-lint/releases/tag/v4.4.1)

> ⚠️  Node 14 will be dropped in a later PR for a future major release. Indeed, Node 14 entered maintenance mode and it has no more active and security support. 

# Checklist

- [x] Update package.json scripts to use pnpm
- [x] Upgrade `ember-template-lint`
- [x] Remove `ember-export-application-global`
- [x] Upgrade Node version used in CI
- [x] Update CI jobs to use pnpm
- [x] Run build, lint and test scripts locally

### CODE REVIEW

File: pnpm-workspace.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+packages:
+  - "ember-autofocus-modifier"
+  - "test-app"
```
[the remark this answers is not in the document]
Comment (vscav) [reply pnpm-workspace.yaml:3]: Yes, we should, the editor is supposed to be configured to do this. Weird it didn't do it automatically at first. Let me fix this

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -8,55 +8,61 @@ on:
...
         uses: actions/setup-node@v3
         with:
-          node-version: 14.x
-          cache: yarn
-      - name: Install Dependencies
-        run: yarn install --frozen-lockfile
+          node-version: 16.x
+      - name: Install pnpm
+        uses: NullVoxPopuli/action-setup-pnpm@v1
+        with:
```
[the remark this answers is not in the document]
Comment (vscav) [reply .github/workflows/ci.yml:28]: I think I tried already, but it was early in the migration process when things were still a bit unstable. I give it another shot

File: test-app/README.md [hunk 3]
```diff
@@ -9,15 +9,15 @@ You will need the following things properly installed on your computer.
 
 * [Git](https://git-scm.com/)
 * [Node.js](https://nodejs.org/)
-* [Yarn](https://yarnpkg.com/)
+* [PNPm](https://pnpm.io/)
```
Comment (JMilosevic810) [inline test-app/README.md:12]: Is this a typo or m should be lower case?

Comment (vscav) [reply test-app/README.md:12]: It's a typo. But now I wonder how we want to write it: `PNPM`, `pnpm` or `Pnpm`?

File: .github/workflows/ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (vscav) [reply .github/workflows/ci.yml]: Done and it's working. Thanks for the suggestion!

File: test-app/README.md (hunk 3 above)
Comment (BlueCutOfficial) [reply test-app/README.md:12]: I would go for `pnpm` the same way we usually write `npm`, lower case is used on https://pnpm.io/ title and docs.

Comment (vscav) [reply test-app/README.md]: Updated with `pnpm`. Thanks @BlueCutOfficial!

**Comment by BlueCutOfficial** [comment]
```
"engines": {
    "node": "14.* || 16.* || >= 18
  },
```
If you had to move the CI to node 16 to install dependencies, does it means the addon is no longer compatible with node 14?

**Comment by vscav** [comment]
@BlueCutOfficial Yes, and I was planning on removing node 14 in a follow-up PR

**Comment by BlueCutOfficial** [comment]
Out of curiosity, is the test-app supposed to display something on serve?

**Comment by vscav** [comment]
@BlueCutOfficial I don't think so. And the master version of the package doesn't display anything neither

**Review by BlueCutOfficial — APPROVED** [review]
Indeed I see no template in the test app hierarchy, just the `index.html`. I could clone the repo and run the test-app correctly with node 16 on this branch, so LGTM

File: test-app/README.md (hunk 3 above)
Comment (JMilosevic810) [reply test-app/README.md]: Actually, on https://pnpm.io the title is different every time. So it looks like any of the formats is correct? 😅

**Review by anas7asia — APPROVED** [review]

**Pull request merged**